File: Pascal-Delphi/Bank/TransactionUnit.pas
unit TransactionUnit;

{$MODE OBJFPC}
{$M+}

interface

type
  TTransaction = class
  private
    FAccountNumber: string;
    FAmount: double;
    FTransactionType: string;
  public
    constructor Create(AccountNumber: string; Amount: double; TransactionType: string);
    function GetAccountNumber: string;
    function GetAmount: double;
    function GetTransactionType: string;
  end;

implementation

{ TTransaction }

constructor TTransaction.Create(AccountNumber: string; Amount: double;
  TransactionType: string);
begin
  FAccountNumber := AccountNumber;
  FAmount := Amount;
  FTransactionType := TransactionType;
end;

function TTransaction.GetAccountNumber: string;
begin
  Result := FAccountNumber;
end;

function TTransaction.GetAmount: double;
begin
  Result := FAmount;
end;

function TTransaction.GetTransactionType: string;
begin
  Result := FTransactionType;
end;

end.

File: C#-Translated/Bank/Transpiler/Program/TransactionUnit.cs
namespace TransactionUnit
{
    public static class Scope
    {
        public class TTransaction
        {
            private string FAccountNumber;
            private double FAmount;
            private string FTransactionType;
            public TTransaction(string AccountNumber, double Amount, string TransactionType)
            {
                FAccountNumber = AccountNumber;
                FAmount = Amount;
                FTransactionType = TransactionType;
            }

            public string GetAccountNumber()
            {
                string Result;
                Result = FAccountNumber;
                return Result;
            }

            public double GetAmount()
            {
                double Result;
                Result = FAmount;
                return Result;
            }

            public string GetTransactionType()
            {
                string Result;
                Result = FTransactionType;
                return Result;
            }
        }
    }
}